### PR TITLE
Chore/spectacle mdx loader addition

### LINKED ISSUE #1164: Infra: Bring `spectacle-mdx-loader` in to the monorepo
- [ ] Add `packages/spectacle-mdx-loader`
- [ ] Add `examples/mdx` that uses the loader
- [ ] Remove old repository
- [ ] Verify we can correctly publish.

### PR DESCRIPTION
**Pull request by paulmarsicloud**
### Description

First pass and attempt at bringing in `spectacle-mdx-loader` into the monorepo. Took onboard @ryan-roemer's advice to remove all devDeps and scripts and focus on getting `lint` and `prettier` working as a start. Moved `prop-types` and `spectacle-mdx-loader` into top-level `package.json` so that `spectacle-mdx-loader` could be called directly from `examples/mdx/webpack.config.js`.

In a good spot now for a second pair of eyes to take a look through 👍 

Fixes #1164 

#### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

### Checklist: (Feel free to delete this section upon completion)

- [ ] I have included a [changeset](../CONTRIBUTING.md#changesets) if this change will require a version change to one of the packages.
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have run `pnpm run check:ci` and all checks pass
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by fritz-c — CHANGES_REQUESTED** [review]

File: packages/spectacle-mdx-loader/LICENSE.txt
Comment (ryan-roemer) [reply packages/spectacle-mdx-loader/LICENSE.txt:1]: Ooh, I noticed we don't ship `LICENSE.txt` in `create-spectacle` and `spectacle`. We should keep this here and _add_ them to the other packages we ship.

cc @gksander @carlos-kelly

File: packages/spectacle-mdx-loader/CONTRIBUTING.md
Comment (ryan-roemer) [reply packages/spectacle-mdx-loader/CONTRIBUTING.md:9]: Yeah, we should remove this file and integrate with root `CONTRIBUTING.md` guide.

File: examples/mdx/webpack.config.js
Comment (ryan-roemer) [reply examples/mdx/webpack.config.js:12]: Hmmm -- not sure here. The root babel config has TS and stuff we don't need. For here we only need enough to support the demo. Could try just removing this field and see if it still works.

File: package.json
Comment (ryan-roemer) [inline package.json:41]: These deps should be in `examples/mdx/package.json`

File: package.json
Comment (ryan-roemer) [inline package.json:174]: GLOBAL: We should add all the `"./packages/spectacle-mdx-loader:*"` here on all the applicable wireit tasks.

File: examples/mdx/index.js
Comment (paulmarsicloud) [reply examples/mdx/index.js:45]: That I am going to have to refer to the experts on (@ryan-roemer / @carlos-kelly ) 😄

File: examples/mdx/slides.mdx
Comment (paulmarsicloud) [reply examples/mdx/slides.mdx]: updated!

File: examples/mdx/test-component.js
Comment (paulmarsicloud) [reply examples/mdx/test-component.js:2]: this guy? https://github.com/FormidableLabs/spectacle/blob/chore/spectacle-mdx-loader-addition/examples/mdx/package.json#L11

File: examples/mdx/slides.mdx
Comment (paulmarsicloud) [reply examples/mdx/slides.mdx]: I'm not familiar either, but I changed it to `jsx` and it worked!

File: examples/mdx/webpack.config.js
Comment (paulmarsicloud) [reply examples/mdx/webpack.config.js]: I've removed the option altogether and it still ran fine to support the demo. If you would like me to add certain things back, I'm happy to do so!

File: packages/spectacle-mdx-loader/CONTRIBUTING.md
Comment (paulmarsicloud) [reply packages/spectacle-mdx-loader/CONTRIBUTING.md]: Removed!

File: packages/spectacle-mdx-loader/README.md
Comment (paulmarsicloud) [reply packages/spectacle-mdx-loader/README.md]: updated!

File: package.json
Comment (paulmarsicloud) [reply package.json]: these have been moved 👍

File: package.json
Comment (paulmarsicloud) [reply package.json:173]: I've updated these, let me know if I've missed anything!

File: packages/spectacle-mdx-loader/src/helpers.js
Comment (paulmarsicloud) [reply packages/spectacle-mdx-loader/src/helpers.js:15]: this too I'm going to have to defer to the expert (@ryan-roemer) 😄

Comment (ryan-roemer) [reply packages/spectacle-mdx-loader/src/helpers.js:15]: Let's just add a ticket to add tests to `spectacle-mdx-loader` as we have none.

File: examples/mdx/index.js
Comment (ryan-roemer) [reply examples/mdx/index.js:45]: Good find @fritz-c 

@paulmarsicloud it seems to be a long gone option. Maybe remove `loop` and re-check everything works?

File: examples/mdx/test-component.js
Comment (fritz-c) [reply examples/mdx/test-component.js:2]: Oh, ok. 👍

File: examples/mdx/package.json
Comment (ryan-roemer) [inline examples/mdx/package.json:11]: I think this should be a prod dep?

File: examples/mdx/package.json
Comment (ryan-roemer) [inline examples/mdx/package.json:10]: I think this one too should be a prod dep?

File: examples/mdx/webpack.config.js
Comment (ryan-roemer) [inline examples/mdx/webpack.config.js:36]: We need a `devDependencies` for this.

File: examples/mdx/webpack.config.js
Comment (ryan-roemer) [inline examples/mdx/webpack.config.js:8]: We need a `devDependencies` for this (technically in root, but should have here if we're not getting config from root).

File: examples/mdx/index.js
Comment (ryan-roemer) [reply examples/mdx/index.js]: I removed this.

**Comment by ryan-roemer** [comment]
@paulmarsicloud I've finished my cleanup of examples configs and dependencies. Mind double checking my commits?

@carlos-kelly @gksander @fritz-c I've noticed a discrepancy in code blocks in MD vs. MDX.

For our MD example with `pnpm start:md` for the code page we have:

Screen Shot 2022-09-21 at 1 50 03 PM

For our MDX example with `pnpm start:mdx` for the code page with exact same sample we have:

Screen Shot 2022-09-21 at 1 50 08 PM

Any ideas here? Thanks!

**Comment by ryan-roemer** [comment]
Offhand, now that I'm looking at things looks like our `Markdown` component does a **lot** more than `mdxComponentMap` does alone -- and we recommend in our guide to just use the `mdxComponentMap` for MDX decks.

**Comment by ryan-roemer** [comment]
Per slack discussion, we're not blocking on this and I've filed https://github.com/FormidableLabs/spectacle/issues/1232 to capture.

**Review by ryan-roemer — APPROVED** [review]
Great work Paul! 🎉

**Comment by paulmarsicloud** [comment]
> @paulmarsicloud I've finished my cleanup of examples configs and dependencies. Mind double checking my commits?

Commits look good and works proper on my end!

File: packages/spectacle-mdx-loader/src/helpers.js
Comment (ryan-roemer) [reply packages/spectacle-mdx-loader/src/helpers.js:15]: Added https://github.com/FormidableLabs/spectacle/issues/1235.

**Comment by ryan-roemer** [comment]
@paulmarsicloud We're missing a `build` task for `examples/mdx` which I'm going to add (and to CI)

**Comment by ryan-roemer** [comment]
I've added the `build` task. I think we're otherwise ready for this PR with all the rest of the follow-on work ticketed.

**Pull request merged**
